ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud.azure;
+
+import com.github.ambry.cloud.CloudDestination;
+import com.github.ambry.commons.BlobId;
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.OperationContext;
+import com.microsoft.azure.storage.blob.BlobContainerPublicAccessType;
+import com.microsoft.azure.storage.blob.BlobRequestOptions;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import java.io.InputStream;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Implementation of {@link CloudDestination} that interacts with Azure Blob Storage service.
+ */
+class AzureCloudDestination implements CloudDestination {
```

Comment:
Better to have a test for it.

Reply:
I would love to, but there's no easy way to mock final classes which is what Azure uses. There is hidden mockito support for it but I can't get it to build.

Reply:
How about pass in a good configSpec and then test it?

Reply:
The config spec is the Azure storage key, and I can't store it plaintext in the code.

Reply:
I added the test class.  It passes in OS branch, we'll have to see if it also passes in the Ambry MP with the LI gradle config.